### DISCUSSION TITLE
%EA%B5%90%EC%9C%A1-%EC%82%B0%EC%97%85%EC%97%90-%EB%8C%80%ED%95%9C-%EA%B3%A0%EC%B0%B0/

### DISCUSSION THREAD

**Discussion by giscus** [heli-os/profile-giscus#12] [Comments]
# %EA%B5%90%EC%9C%A1-%EC%82%B0%EC%97%85%EC%97%90-%EB%8C%80%ED%95%9C-%EA%B3%A0%EC%B0%B0/

볼타 CTO 테오 블로그

https://dataportal.kr/%EA%B5%90%EC%9C%A1-%EC%82%B0%EC%97%85%EC%97%90-%EB%8C%80%ED%95%9C-%EA%B3%A0%EC%B0%B0/

**Comment by Flamme1004K** [comment]
이 연습을 게을리 하다보면 1년, 2년이 지났을 때에는 열심히 하고 있지만 성장하지 않는 것 같다라는 생각을 하게 될 것이고 3년, 4년이 지났을 때에는 내가 모르는 것이 너무 많다라는 생각이 들 것입니다.

최근에 느꼇던 감정이네요.  매우 공감합니다.

아직까지도, 저는 학습을 위한 연습에 대해 어떻게 해야할지 많은 고민을 가지고 있는데, 테오님은 어떻게 방향성을 잡으셨나요?

**Reply by heli-os** [reply]
저는 아는 것과 모르는 것의 경계를 명확히 나누기 위해 의도적으로 신경 쓰는 편입니다.

기술은 공부하다 보면 그 방법이 정말 다양한 편인데요. 특히 인기 있는 기술은 이미 다른 사람들이 많이 공부하고 그 노하우를 이곳저곳 공유해놓기도 합니다. 이때 많이 실수하는 부분이 그것들을 먼저 살펴보는 행위입니다. 처음 빠르게 파악하고 실무에서 적용하는 데 효율적이긴 하지만, 가장 큰 문제는 `그래서 내가 어느 정도 아는 거지?`라는 의문이 생길 수밖에 없습니다.

그래서 저는 아래와 같은 방식을 주변에 권하고 있습니다.

1. 공부하고자 하는 것을 최소로 파악합니다. 공식 문서를 가장 권장합니다.
2. 공식 문서를 보고 실제 구현이나 더 자세한 내용이 궁금하다면 실제 코드의 구현을 찾아보지 않고, 직접 만들어봅니다.
  2-1. 어떤 요소가 필요할지,
  2-2. 문서에는 담겨있지 않은 엣지 케이스는 없을지,
  2-3. ...
  2-4. 이 과정에서 문서에 담겨 있던 것(4)에서 내 생각(2)을 더 해가며, 6을 명확히 내 것으로 만들 수 있습니다.
3. 그 후 실제 구현과 기존의 레퍼런스와 비교해 봅니다.
4. 만약 비교해 보았을 때,
  4-1. 기존 레퍼런스가 '8'만큼 담겨있다면, 6까지는 명확히 아는 것이니 7과 8만 추가로 공부하면 됩니다.
  4-2. 반대로 기존 레퍼런스가 '5'만큼 담겨있다면, 초과하는 1만큼 기여할 수 있게 됩니다.

이런 학습 루틴을 가지고 있으면, 계단식으로, 점진적으로 성장한다는 느낌을 스스로 받을 수 있습니다.

이 과정의 앞과 뒤, 그리고 사이사이에 필요한 더 중요한 것들도 있지만.. 우선 이 정도 이야기를 드릴 수 있을 것 같습니다.

**Reply by Flamme1004K** [reply]
감사합니다! 테오님, 권해주신 방식으로 학습을 해봐야 되겠네요!
